AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/InitialSync/InitialSyncOrchestrator.swift

```diff
@@ -85,39 +97,56 @@ final class AWSInitialSyncOrchestrator: InitialSyncOrchestrator {
 
     /// Enqueues sync operations for models and downstream dependencies
     private func enqueueSyncOperation(for modelType: Model.Type) {
-        let syncOperationCompletion: SyncOperationResultHandler = { result in
-            if case .failure(let dataStoreError) = result {
-                let syncError = DataStoreError.sync(
-                    "An error occurred syncing \(modelType.modelName)",
-                    "",
-                    dataStoreError)
-                self.syncErrors.append(syncError)
-            }
-        }
-
         let initialSyncForModel = InitialSyncOperation(modelType: modelType,
                                                        api: api,
                                                        reconciliationQueue: reconciliationQueue,
                                                        storageAdapter: storageAdapter,
-                                                       dataStoreConfiguration: dataStoreConfiguration,
-                                                       completion: syncOperationCompletion)
+                                                       dataStoreConfiguration: dataStoreConfiguration)
+
+        initialSyncOperationSinks[modelType.modelName] = initialSyncForModel
+            .publisher
+            .receive(on: concurrencyQueue)
+            .sink(receiveCompletion: { result in
+                if case .failure(let dataStoreError) = result {
+                    let syncError = DataStoreError.sync(
+                        "An error occurred syncing \(modelType.modelName)",
+                        "",
+                        dataStoreError)
+                    self.syncErrors.append(syncError)
+                }
+                self.initialSyncOperationSinks.removeValue(forKey: modelType.modelName)
+                self.onReceiveCompletion()
+            }, receiveValue: onReceiveValue(_:))
 
         syncOperationQueue.addOperation(initialSyncForModel)
     }
 
-    private func resolveCompletion() {
-        // TODO: Invoke error callback for sync errors
+    private func onReceiveValue(_ value: InitialSyncOperationEvent) {
+        initialSyncOrchestratorTopic.send(value)
+    }
+
+    private func onReceiveCompletion() {
+        guard initialSyncOperationSinks.isEmpty else {
+            return
+        }
+
+        let completionResult = makeCompletionResult()
+        if case .success = completionResult {
+            initialSyncOrchestratorTopic.send(completion: .finished)
```

Comment:
What about the error case? You still need to complete the publisher in that case right?